portable/IAR/AtmelSAM7S64/lib_AT91SAM7X256.h

```diff
@@ -1,1469 +1,1396 @@
+/**---------------------------------------------------------------------------- */
+    __inline void AT91F_PDC_EnableRx( AT91PS_PDC pPDC ) /* \arg pointer to a PDC controller */
+    {
+        pPDC->PDC_PTCR = AT91C_PDC_RXTEN;
+    }
+
+/**---------------------------------------------------------------------------- */
+/** \fn    AT91F_PDC_DisableTx */
+/** \brief Disable transmit */
+/**---------------------------------------------------------------------------- */
+    __inline void AT91F_PDC_DisableTx( AT91PS_PDC pPDC ) /* \arg pointer to a PDC controller */
+    {
+        pPDC->PDC_PTCR = AT91C_PDC_TXTDIS;
+    }
+
+/**---------------------------------------------------------------------------- */
+/** \fn    AT91F_PDC_DisableRx */
+/** \brief Disable receive */
+/**---------------------------------------------------------------------------- */
+    __inline void AT91F_PDC_DisableRx( AT91PS_PDC pPDC ) /* \arg pointer to a PDC controller */
+    {
+        pPDC->PDC_PTCR = AT91C_PDC_RXTDIS;
+    }
+
+/**---------------------------------------------------------------------------- */
+/** \fn    AT91F_PDC_IsTxEmpty */
+/** \brief Test if the current transfer descriptor has been sent */
+/**---------------------------------------------------------------------------- */
+    __inline int AT91F_PDC_IsTxEmpty( /* \return return 1 if transfer is complete */
+        AT91PS_PDC pPDC )             /* \arg pointer to a PDC controller */
+    {
+        return !( pPDC->PDC_TCR );
+    }
+
+/**---------------------------------------------------------------------------- */
+/** \fn    AT91F_PDC_IsNextTxEmpty */
+/** \brief Test if the next transfer descriptor has been moved to the current td */
+/**---------------------------------------------------------------------------- */
+    __inline int AT91F_PDC_IsNextTxEmpty( /* \return return 1 if transfer is complete */
+        AT91PS_PDC pPDC )                 /* \arg pointer to a PDC controller */
+    {
+        return !( pPDC->PDC_TNCR );
+    }
+
+/**---------------------------------------------------------------------------- */
+/** \fn    AT91F_PDC_IsRxEmpty */
+/** \brief Test if the current transfer descriptor has been filled */
+/**---------------------------------------------------------------------------- */
+    __inline int AT91F_PDC_IsRxEmpty( /* \return return 1 if transfer is complete */
+        AT91PS_PDC pPDC )             /* \arg pointer to a PDC controller */
+    {
+        return !( pPDC->PDC_RCR );
+    }
+
+/**---------------------------------------------------------------------------- */
+/** \fn    AT91F_PDC_IsNextRxEmpty */
+/** \brief Test if the next transfer descriptor has been moved to the current td */
+/**---------------------------------------------------------------------------- */
+    __inline int AT91F_PDC_IsNextRxEmpty( /* \return return 1 if transfer is complete */
+        AT91PS_PDC pPDC )                 /* \arg pointer to a PDC controller */
```

Comment:
Should this be one line?

Reply:
Answered [here](https://github.com/FreeRTOS/FreeRTOS-Kernel/pull/768#discussion_r1313523527)